service/history/outbound_queue_factory.go

```diff
@@ -127,7 +132,16 @@ func NewOutboundQueueFactory(params outboundQueueFactoryParams) QueueFactory {
 					SchedulerFactory: func(
 						key queues.StateMachineTaskTypeNamespaceIDAndDestination,
 					) ctasks.RunnableScheduler {
-						return ctasks.NewDynamicWorkerPoolScheduler(groupLimiter{key})
+						nsName, err := params.NamespaceRegistry.GetNamespaceName(namespace.ID(key.NamespaceID))
+						if err != nil {
+							nsName = namespace.Name(key.NamespaceID)
+						}
```

Comment:
Let's propagate this error instead of silently doing the wrong thing.

Reply:
Also note that if the namespace is renamed, we don't reapply the limiter settings because the scheduler is cached by namespace ID.